### ISSUE TITLE
Importing torch after TensorFlow results in std::runtime_error

### ISSUE THREAD

**Issue by Dobiasd** [pytorch/pytorch#101152]
### 🐛 Describe the bug

The following minimal example (`Dockerfile`, just run `docker build` with it ) reproduces the error:

```Dockerfile
FROM python:3.11.3
RUN pip install tensorflow==2.12.0 torch==2.0.1
RUN python -c "import tensorflow as tf; import torch"
```

```
terminate called after throwing an instance of 'std::runtime_error'
  what():  random_device could not be read
Aborted (core dumped)
```

It does not happen when importing torch first.

### Versions

```
Collecting environment information...
PyTorch version: 2.0.1+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Debian GNU/Linux 11 (bullseye) (x86_64)
GCC version: (Debian 10.2.1-6) 10.2.1 20210110
Clang version: Could not collect
CMake version: version 3.26.3
Libc version: glibc-2.31

Python version: 3.11.3 (main, May  4 2023, 05:53:32) [GCC 10.2.1 20210110] (64-bit runtime)
Python platform: Linux-5.15.0-71-generic-x86_64-with-glibc2.31
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   39 bits physical, 48 bits virtual
CPU(s):                          4
On-line CPU(s) list:             0-3
Thread(s) per core:              1
Core(s) per socket:              4
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           94
Model name:                      Intel(R) Core(TM) i5-6600 CPU @ 3.30GHz
Stepping:                        3
CPU MHz:                         3300.000
CPU max MHz:                     3900.0000
CPU min MHz:                     800.0000
BogoMIPS:                        6599.98
Virtualization:                  VT-x
L1d cache:                       128 KiB
L1i cache:                       128 KiB
L2 cache:                        1 MiB
L3 cache:                        6 MiB
NUMA node0 CPU(s):               0-3
Vulnerability Itlb multihit:     KVM: Mitigation: VMX disabled
Vulnerability L1tf:              Mitigation; PTE Inversion; VMX conditional cache flushes, SMT disabled
Vulnerability Mds:               Mitigation; Clear CPU buffers; SMT disabled
Vulnerability Meltdown:          Mitigation; PTI
Vulnerability Mmio stale data:   Mitigation; Clear CPU buffers; SMT disabled
Vulnerability Retbleed:          Mitigation; IBRS
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:             Mitigation; Microcode
Vulnerability Tsx async abort:   Mitigation; TSX disabled
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault invpcid_single pti ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx rdseed adx smap clflushopt intel_pt xsaveopt xsavec xgetbv1 xsaves dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp md_clear flush_l1d arch_capabilities

Versions of relevant libraries:
[pip3] numpy==1.23.5
[pip3] torch==2.0.1
[pip3] triton==2.0.0
[conda] Could not collect
```

cc @peterjc123 @mszhanyi @skyline75489 @nbcsm @vladimir-aubrecht

**Comment by AdonaiVera** [comment]
I faced a similar problem while running my Instance on an EC2 server with Ubuntu 20.04. I have both tensorflow==2.12.0 and torch==2.0.1 installed, and I am running two object detection models. I am using an ensemble model. 

To solve the issue, I imported torch first, and it worked.

**Comment by davidpanic** [comment]
I'm seeing this happen in Kubernetes when the pod doesn't have `tty: true` set.
I don't think that there's a way to allocate TTYs in docker build stages.

**Comment by Dobiasd** [comment]
@iremyux I'm using Docker on Ubuntu. 🐧

**Comment by iremyux** [comment]
Hello, I have removed the 'module:windows' label from this issue as it does not appear to be relevant. I made an effort to assign a more accurate label, but please feel free to modify or remove it if you believe it is not suitable.

**Comment by animeesh** [comment]
hello, I have put a torch, TensorFlow respectively in my requirements folder in windows os .
facing a similar issue.
terminate called after throwing an instance of 'std::runtime_error'
  what():  random_device could not be read

**Comment by Dobiasd** [comment]
I just tested with the latest versions of `tensorflow` and `torch`, and the problem no longer exists. I.e., the following `Dockerfile` works without error. :tada:

```Dockerfile
FROM python:3.11.3
RUN pip install tensorflow==2.16.1 torch==2.3.0
RUN python -c "import tensorflow as tf; import torch"
```
